test/Libraries/Microsoft.Extensions.AI.AotCompatibility.TestApp/Microsoft.Extensions.AI.AotCompatibility.TestApp.csproj

```diff
@@ -0,0 +1,36 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFrameworks>net9.0</TargetFrameworks>
+    <PublishAot>true</PublishAot>
+    <TrimmerSingleWarn>false</TrimmerSingleWarn>
+    <SelfContained>true</SelfContained>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```

Comment:
Is this not already the case for this repo?

Reply:
Nope, only in CI builds. So we've just enabled it always for the M.E.AI projects.

Reply:
We did it as this repo has an unusual number of analyzers, some of which require things done outside of the source files (like the public api anaylzer and the API stage analyzer). After some feedback, devs working on the repo and experimenting wanted the ability to turn these off for innerloop, and then just relying on passing a flag when this should be enabled. Then we made a CI leg specific to "Correctness" which ensures this flag is set. IIRC we copied this pattern from dotnet/roslyn.